src/station_cmd.cpp

```diff
@@ -1324,6 +1205,224 @@ void SetRailStationTileFlags(TileIndex tile, const StationSpec *statspec)
+		}
+	} else {
+		return CommandCost(INVALID_STRING_ID);
+
+		if (GetTileMaxZ(tile) + (drive_through ? 1 : 2) > bridge_height) {
+			return CommandCost(STR_ERROR_BRIDGE_TOO_LOW_FOR_STATION);
+		}
+	}
+
+	BridgePiecePillarFlags disallowed_pillar_flags = (BridgePiecePillarFlags) 0;
+	if (spec != nullptr && spec->internal_flags.Test(RoadStopSpecIntlFlag::BridgeDisallowedPillarsSet)) {
+		disallowed_pillar_flags = (BridgePiecePillarFlags) spec->bridge_disallowed_pillars[drive_through ? (GFX_TRUCK_BUS_DRIVETHROUGH_OFFSET + DiagDirToAxis(entrance)) : entrance];
+	} else if (drive_through) {
+		disallowed_pillar_flags = (BridgePiecePillarFlags) (DiagDirToAxis(entrance) == AXIS_X ? 0x50 : 0xA0);
+	} else {
+		SetBit(disallowed_pillar_flags, 4 + entrance);
+	}
+	if ((GetBridgeTilePillarFlags(tile, northern_bridge_end, southern_bridge_end, bridge_type, bridge_transport_type) & disallowed_pillar_flags) == 0) {
+		return CommandCost();
+	} else {
+		return CommandCost(STR_ERROR_BRIDGE_PILLARS_OBSTRUCT_STATION);
+	}
+}
+
+/**
+ * Checks if a road stop can be built at the given tile.
+ * @param cur_tile Tile to check.
+ * @param allowed_z Height allowed for the tile. If allowed_z is negative, it will be set to the height of this tile.
+ * @param flags Operation to perform.
+ * @param invalid_dirs Prohibited directions (set of DiagDirections).
+ * @param is_drive_through True if trying to build a drive-through station.
+ * @param station_type Station type (bus, truck or road waypoint).
+ * @param axis Axis of a drive-through road stop.
+ * @param station StationID to be queried and returned if available.
+ * @param rt Road type to build, may be INVALID_ROADTYPE if an existing road is required.
+ * @return The cost in case of success, or an error code if it failed.
+ */
+static CommandCost CheckFlatLandRoadStop(TileIndex cur_tile, int &allowed_z, const RoadStopSpec *spec, DoCommandFlags flags, DiagDirections invalid_dirs, bool is_drive_through, StationType station_type, Axis axis, StationID *station, RoadType rt)
+{
+	CommandCost cost(EXPENSES_CONSTRUCTION);
+
+	bool allow_under_bridge = spec != nullptr && spec->internal_flags.Test(RoadStopSpecIntlFlag::BridgeHeightsSet);
+	CommandCost ret = CheckBuildableTile(cur_tile, invalid_dirs, allowed_z, !is_drive_through, true);
+	if (ret.Failed()) return ret;
+	cost.AddCost(ret.GetCost());
+
+	if (allow_under_bridge && IsBridgeAbove(cur_tile)) {
+		TileIndex southern_bridge_end = GetSouthernBridgeEnd(cur_tile);
+		TileIndex northern_bridge_end = GetNorthernBridgeEnd(cur_tile);
+		CommandCost bridge_ret = IsRoadStopBridgeAboveOK(cur_tile, spec, is_drive_through, DiagDirection::DIAGDIR_NE /*obviously wrong, but how do you get the "first bit" from invalid_dirs? and how would that be correct?? */,
+														 northern_bridge_end, southern_bridge_end, GetBridgeHeight(southern_bridge_end),
+														 GetBridgeType(southern_bridge_end), GetTunnelBridgeTransportType(southern_bridge_end));
```

Comment:
`FindFirstBit()` normally... whatever that means in this context.

Reply:
that doesn't seem to work on `DiagDirections` ?